demos/smart_framing_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,321 @@
+        for (const auto& el : objects) {
+            if (el.labelID == 0) {//person ID
+                init_rect = init_rect | static_cast<cv::Rect>(el);
+            }
+        }
+        slog::debug << "SF result rect" << init_rect << slog::endl;
+        if (!init_rect.empty()) {
+            cv::Rect even_rect;
+            even_rect.x = (init_rect.x % 2 == 0) ? (init_rect.x) : (init_rect.x - 1);
+            even_rect.y = (init_rect.y % 2 == 0) ? (init_rect.y) : (init_rect.y - 1);
+            even_rect.width = (init_rect.width % 2 == 0) ? (init_rect.width) : (init_rect.width - 1);
+            even_rect.height = (init_rect.height % 2 == 0) ? (init_rect.height) : (init_rect.height - 1);
+
+            cv::Mat SF_ROI;
+            image(even_rect).copyTo(SF_ROI);
+
+            cv::Mat SF_resized_ROI;
+            cv::Size target_size;
+            target_size.height = image.size().height;
+            target_size.width = even_rect.width * (static_cast<float>(image.size().height) / static_cast<float>(even_rect.height));
+            target_size.width = (target_size.width % 2 == 0) ? (target_size.width) : (target_size.width - 1);
+            if (target_size.width > image.size().width) {
+                target_size.width = image.size().width;
+            }
+            cv::resize(SF_ROI, SF_resized_ROI, target_size);
+            int left_add = (image.size().width - target_size.width) / 2;
+            int right_add = (image.size().width - target_size.width) / 2;
+            cv::copyMakeBorder(SF_resized_ROI, out,
+                0,
+                0,
+                left_add,
+                right_add,
+                cv::BORDER_CONSTANT, cv::Scalar::all(127));
+        } else {
+            image.copyTo(out);
+        }
+    }
+};
+
+GAPI_OCV_KERNEL(OCVSuperResolutionPostProcessingKernel, custom::GSuperResolutionPostProcessingKernel) {
+    static void run(const cv::Mat & image, cv::Mat & out) {
+        float* outputData = (float*)image.data;
+        size_t outChannels = image.size[1];
+        size_t outHeight = image.size[2];
+        size_t outWidth = image.size[3];
+        size_t numOfPixels = outWidth * outHeight;
+        std::vector<cv::Mat> imgPlanes;
+        if (outChannels == 3) {
+            imgPlanes = std::vector<cv::Mat>{
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[0])),
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[numOfPixels])),
+                  cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[numOfPixels * 2])) };
+        }
+        else {
+            imgPlanes = std::vector<cv::Mat>{ cv::Mat(outHeight, outWidth, CV_32FC1, &(outputData[0])) };
+            // Post-processing for text-image-super-resolution models
+            cv::threshold(imgPlanes[0], imgPlanes[0], 0.5f, 1.0f, cv::THRESH_BINARY);
+        }
+        for (auto& img : imgPlanes)
+            img.convertTo(img, CV_8UC1, 255);
+
+        cv::merge(imgPlanes, out);
+    }
+};
```

Comment:
Can we unroll this kernel into a graph is possible? I see its `treshold`/`convertTo`/`merge` path can run on Fluid as well.